src/pkg/package/lockfile/io.go

```diff
@@ -0,0 +1,133 @@
+package lockfile
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+
+	"github.com/Southclaws/sampctl/src/pkg/infrastructure/print"
+	"github.com/Southclaws/sampctl/src/pkg/infrastructure/util"
+)
+
+// Load attempts to load a lockfile from the given directory.
+// Format is auto-detected from file content. Returns nil if no lockfile exists.
+func Load(dir string) (*Lockfile, error) {
+	path := filepath.Join(dir, Filename)
+
+	if !util.Exists(path) {
+		return nil, nil
+	}
+
+	return loadFromFile(path)
+}
+
+// loadFromFile loads a lockfile from a specific file path.
+// Format is auto-detected from content (JSON starts with '{', otherwise YAML).
+func loadFromFile(path string) (*Lockfile, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read lockfile")
+	}
+
+	var lockfile Lockfile
+
+	// Auto-detect format: JSON starts with '{', otherwise try YAML
+	if len(data) > 0 && data[0] == '{' {
```

Comment:
This not needed, the lock file should just be .json as it's not something people are going to be looking at much.